renderdoc/driver/gl/gl_manager.cpp

```diff
@@ -608,8 +609,22 @@ void GLResourceManager::PrepareTextureInitialContents(ResourceId liveid, Resourc
 
     if(HasExt[ARB_texture_swizzle] || HasExt[EXT_texture_swizzle])
     {
-      gl.glGetTextureParameterivEXT(res.name, details.curType, eGL_TEXTURE_SWIZZLE_RGBA,
-                                    (GLint *)&state->swizzle[0]);
+      if(IsGLES)
```

Comment:
Yes, I know they exist, but I just thought that if the RGBA version is available, then I use it.

I have some problems with your recommendation. First of all, where should I put these functions? Just because it is also used from `gl_resources`'s `EmulateLuminanceFormat` function, where only `GLHookSet` is available, but `WrappedOpenGL` is not. Or should I add `GLHookSet` as an additional parameter for these funcions? Do we really need this? There are many code parts which are cloned in several places...